src/resource/page/ResourcePeriodsCopyFieldset.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { Notification, IconCopy } from 'hds-react';
+import api from '../../common/utils/api/api';
+import {
+  formatDate,
+  datetimeFormFormat,
+} from '../../common/utils/date-time/format';
+import { PrimaryButton } from '../../components/button/Button';
+import toast from '../../components/notification/Toast';
+
+export type TargetResourcesProps = {
+  mainResourceName?: string;
+  mainResourceId?: number;
+  targetResources?: string[];
+  modified?: string;
+};
+
+export default function ResourcePeriodsCopyFieldset({
+  mainResourceName,
+  mainResourceId,
+  targetResources = [],
+  onChange,
+  modified,
+  hasReferrer,
+}: TargetResourcesProps & {
+  hasReferrer: boolean;
+  onChange: (value: TargetResourcesProps | undefined) => void;
+}): JSX.Element {
+  const [isCopyLoading, setIsCopyLoading] = useState<boolean>(false);
+
+  const copyDatePeriods = async (): Promise<void> => {
+    setIsCopyLoading(true);
+
+    if (!mainResourceId || targetResources.length === 0) {
+      return;
+    }
+
+    try {
+      await api.copyDatePeriods(mainResourceId, targetResources);
+      toast.success({
+        dataTestId: 'period-copy-success',
+        label: 'Aukiolotietojen kopiointi onnistui',
+        text: 'Voit tarvittaessa kopioida aukiolotiedot uudelleen',
+      });
+      onChange({
+        mainResourceName,
+        mainResourceId,
+        targetResources,
+        modified: new Date().toJSON(),
+      });
+      setIsCopyLoading(false);
+      if (hasReferrer) {
+        window.close();
+      }
+    } catch (err) {
+      toast.error({
+        dataTestId: 'period-copy-error',
+        label: 'Aukiolotietojen kopointi epäonnistui',
+        text: 'Yritä myöhemmin uudelleen',
+      });
+      setIsCopyLoading(false);
+
+      // eslint-disable-next-line no-console
+      console.error(err); // For debug purposes
+    }
+  };
+
+  return (
+    <div className="resource-copy-date-periods">
+      <Notification
+        type="alert"
+        label={`Olet valinnut joukkopäivityksessä ${
+          (targetResources?.length || 0) + 1
+        } pistettä. Klikkasit "${mainResourceName}"n aukiolotietoa. Sinulle on auennut ”${mainResourceName}”n aukiolotieto muokattavaksi.`}>
```

Comment:
This sounds pretty TPREK-specific. Are we planning to make the feature tprek only? I guess we can update later if need to.